src/config.cpp

```diff
@@ -68,7 +71,32 @@ const Config& config()
 {
     return *instance;
 }
+int get_default_mmap_value()
```

Comment:
Please move the function to `util.cpp` with all the other `/proc` / `sysconf` wrappers. I would suggest to name it there `get_perf_event_mlock` and return a `size_t` value in bytes, or in case of an exception a default of 516*1024.